mex/backend/graph/cypher/fetch_merged_items.cql

```diff
@@ -0,0 +1,69 @@
+<# Fetch merged items including their nested objects and referenced identifiers.
```

Comment:
nicely done! this was quite complex i imagine, but looks good. i created a follow-up in MX-1693 so we can try to combat some of the repetition across all queries

Reply:
Thank you 😊 I certainly learned some cypher along the way.